grpc/conn.go

```diff
@@ -0,0 +1,67 @@
+package grpc
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"go.viam.com/utils/rpc"
+	googlegrpc "google.golang.org/grpc"
+)
+
+// ReconfigurableClientConn allows for the underlying client connections to be swapped under the hood.
+type ReconfigurableClientConn struct {
+	connMu sync.RWMutex
+	conn   rpc.ClientConn
+}
+
+// Invoke invokes using the underlying client connection.
+func (c *ReconfigurableClientConn) Invoke(
+	ctx context.Context,
+	method string,
+	args, reply interface{},
+	opts ...googlegrpc.CallOption,
+) error {
+	c.connMu.RLock()
+	conn := c.conn
+	c.connMu.RUnlock()
+	if conn == nil {
+		return errors.New("not connected")
+	}
+	return conn.Invoke(ctx, method, args, reply, opts...)
```

Comment:
talking to @dgottlieb, there is a concern with the ownership of the connection (who closes it and when) and also the possibility of issues if the connection gets closed while a request is ongoing. With exposing this struct publicly, there is a higher chance of this being misused. I'll add some more documentation, but also think about/test utilizing locks around these calls.

Reply:
after investigating, I don't think it works well to place the Invoke call within the lock - in order to cancel ongoing calls and swap the connection, we'll have to keep a cancel func for each call that is being made, and then cancel each of them when swapping. The placement of the lock would also be quite awkward, maybe we'll need multiple locks (one for the call and one to keep track of the cancels). This makes the overhead bigger, so since this currently works, am opting to keep this implementation until we run into isses

Reply:
> am opting to keep this implementation until we run into isses

That seems fine to me with the new documentation.